---
title: "Pertussis Mini Project"
author: 'Ari_Fon (PID: A15390446)'
date: "3/8/2022"
output: pdf_document
---

# Investigate Pertussis case numbers over time in the US

The CDC has tracked cases numbers since the early 1920s.
https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html


```{r}
cdc <- data.frame(
                                 Year = c(1922L,1923L,1924L,1925L,
                                          1926L,1927L,1928L,1929L,1930L,1931L,
                                          1932L,1933L,1934L,1935L,1936L,
                                          1937L,1938L,1939L,1940L,1941L,1942L,
                                          1943L,1944L,1945L,1946L,1947L,
                                          1948L,1949L,1950L,1951L,1952L,
                                          1953L,1954L,1955L,1956L,1957L,1958L,
                                          1959L,1960L,1961L,1962L,1963L,
                                          1964L,1965L,1966L,1967L,1968L,1969L,
                                          1970L,1971L,1972L,1973L,1974L,
                                          1975L,1976L,1977L,1978L,1979L,1980L,
                                          1981L,1982L,1983L,1984L,1985L,
                                          1986L,1987L,1988L,1989L,1990L,
                                          1991L,1992L,1993L,1994L,1995L,1996L,
                                          1997L,1998L,1999L,2000L,2001L,
                                          2002L,2003L,2004L,2005L,2006L,2007L,
                                          2008L,2009L,2010L,2011L,2012L,
                                          2013L,2014L,2015L,2016L,2017L,2018L,
                                          2019L),
         No..Reported.Pertussis.Cases = c(107473,164191,165418,152003,
                                          202210,181411,161799,197371,
                                          166914,172559,215343,179135,265269,
                                          180518,147237,214652,227319,103188,
                                          183866,222202,191383,191890,109873,
                                          133792,109860,156517,74715,69479,
                                          120718,68687,45030,37129,60886,
                                          62786,31732,28295,32148,40005,
                                          14809,11468,17749,17135,13005,6799,
                                          7717,9718,4810,3285,4249,3036,
                                          3287,1759,2402,1738,1010,2177,2063,
                                          1623,1730,1248,1895,2463,2276,
                                          3589,4195,2823,3450,4157,4570,
                                          2719,4083,6586,4617,5137,7796,6564,
                                          7405,7298,7867,7580,9771,11647,
                                          25827,25616,15632,10454,13278,
                                          16858,27550,18719,48277,28639,32971,
                                          20762,17972,18975,15609,18617)
       )
```

>Q1. Make a figure of cases (y) per year (x) using ggplot 

```{r}
library(tidyverse)
```


```{r}
ggplot(cdc) +
  aes(Year, No..Reported.Pertussis.Cases) +
  geom_point() +
  geom_line() +
  labs(title= "Pertussis Cases by Year (1922-2019", x= "Year", y= "Number of cases")
```

>Q2. Using the ggplot geom_vline() function add lines to your previous plot for the 1946 introduction of the wP vaccine and the 1996 switch to aP vaccine (see example in the hint below). What do you notice?


```{r}
ggplot(cdc) +
  aes(Year, No..Reported.Pertussis.Cases) +
  geom_point() +
  geom_line() +
  labs(title= "Pertussis Cases by Year (1922-2019", x= "Year", y= "Number of cases") +
  geom_vline(xintercept = 1946, color= "blue", linetype= "dashed") +
  geom_vline(xintercept = 1996, color= "red", linetype= "dashed")
  
```

>Q3. Describe what happened after the introduction of the aP vaccine? Do you have a possible explanation for the observed trend?

After the 1946 vaccination the number of cases significantly dropped but after the 1996 vaccination the reported cases went up again. The bacteria could have mutated or evolved as it spread. The new vaccine may not have been as effective as the old one. More sensitive testing may also contributed so we can pick up more cases. Vaccine hesitancy may also have contributed as more people grew wary about vaccines. Vaccine efficacy decreases over times so it is possible that the new vaccine efficacy may decrease faster than the old one. Waning immunity in adolescents is a huge concern in communities. 


Key-point: Despite high levels of acellular pertussis (aP) vaccination, the United States and other countries are now experiencing a significant resurgence in pertussis cases with large outbreaks now once again a major public health concern

Waning of immunity in adolescents originally primed as infants with the newer aP vaccine as compared to the older wP vaccine.



#Exploring CMI-PB data

We will use the **jsonlite** package to read from the CMI-PB database API directly.


```{r}
# Allows us to read, write and process JSON data
library(jsonlite)
```

```{r}
url <- "https://www.cmi-pb.org/api/subject"
subject <- read_json(url, simplifyVector = TRUE)
head(subject, 3)
```

>Q4. How may aP and wP infancy vaccinated subjects are in the dataset?

```{r}
table(subject$infancy_vac)
```
> How many total?

```{r}
nrow(subject)
```


>Q5. How many Male and Female subjects/patients are in the dataset?

```{r}
table(subject$biological_sex)
```


>Q6. What is the breakdown of race and biological sex (e.g. number of Asian females, White males etc…)?

```{r}
table(subject$race, subject$biological_sex)
```
 > Q7 and Q8 are optional.

#Joining Multiple Tables (datasets)

Read the specimen and ab_titer tables into R and store the data as specimen and titer named data frames.

```{r}
# Complete the API URLs...
specimen <- read_json("https://www.cmi-pb.org/api/specimen", simplifyVector = TRUE) 
titer <- read_json("https://www.cmi-pb.org/api/ab_titer", simplifyVector = TRUE)
```


Have a quick look

```{r}
head(specimen, 3)
```


>Q9. Complete the code to join specimen and subject tables to make a new merged data frame containing all specimen records along with their associated subject details:

I need to use inner_join() here.

```{r}
library(dplyr)
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```

>Q10. Now using the same procedure join meta with titer data so we can further analyze this data in terms of time of visit aP/wP, male/female etc.


```{r}
abdata <- inner_join(titer, meta)
dim(abdata)
head(abdata, 3)
```

>Q11. How many specimens (i.e. entries in abdata) do we have for each isotype?

```{r}
table(abdata$isotype)
```

>Q12. What do you notice about the number of visit 8 specimens compared to other visits?

```{r}
table(abdata$visit)
```
Visit 8 is way smaller compared to the other specimens. This is because we are still collecting data which is why it is smaller. Visit is a proxy for time.

# Examine IgG1 Ab titer levels

Now using our joined/merged/linked abdata dataset filter() for IgG1 isotype and exclude the small number of visit 8 entries.

```{r}
ig1 <- abdata %>% filter(isotype == "IgG1", visit!=8)
head(ig1)
```

>Q13. Complete the following code to make a summary boxplot of Ab titer levels for all antigens:

```{r}
library(ggplot2)
ggplot(ig1) +
  aes(ab_titer, antigen) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2)
```


>Q14. What antigens show differences in the level of IgG1 antibody titers recognizing them over time? Why these and not others?

```{r}
ggplot(ig1) +
  aes(ab_titer, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(visit), nrow=2) +
  theme_bw()
```

It seems that FIM2/3, TT, and PTM show differences in the level of IgG1 antibody titers recognizing them over time.This might be because antibodies specifically recognize them. 


>Q15. Filter to pull out only two specific antigens for analysis and create a boxplot for each. You can chose any you like. Below I picked a “control” antigen (“Measles”, that is not in our vaccines) and a clear antigen of interest (“FIM2/3”, extra-cellular fimbriae proteins from B. pertussis that participate in substrate attachment).

```{r}
filter(ig1, antigen=="Measles") %>%
  ggplot() +
  aes(ab_titer, col=infancy_vac) +
  geom_boxplot(show.legend = TRUE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```
It appears to stay relatively consistent at low levels. This is the control after all.

```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(ab_titer, col=infancy_vac) +
  geom_boxplot(show.legend = TRUE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```
FIM2/3 experienced an increase over time and this increase is around the same for both vaccines.In comparison to the control, FM2/3 increases and appears to peak around visit 4.



>Q16. What do you notice about these two antigens time course and the FIM2/3 data in particular?

FIM2/3 in particular had a greater rise over the time course 


>Q17. Do you see any clear difference in aP vs. wP responses?

There doesn't appear to be any clear differences in aP vs, wP respones. In FIM2/3 there was a slight difference between aP and wP but there is overlap so no significant difference.



## Pull RNA-Seq data from the CMI-PB 

We can use the CMI-PB  API to obtain time-cours RNA seq results for wP and aP subjects (i.e. patients)



```{r}
url2 <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url2, simplifyVector = TRUE) 
dim(rna)
```

```{r}
#meta <- inner_join(specimen, subject)
ssrna <- inner_join(rna, meta)
dim(ssrna)
```


>Q18. Make a plot of the time course of gene expression for IGHG1 gene (i.e. a plot of visit vs. tpm).

```{r}
ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)
```
Notice: there is a rise at visit 4

>Q19.: What do you notice about the expression of this gene (i.e. when is it at it’s maximum level)?

It appears that the expression of this gene, i.e. when its at its maximum level, is a little above 6000 tpm at around visit 4.

>Q20. Does this pattern in time match the trend of antibody titer data? If not, why not?

It looks like this pattern in time matches the trend of the antibody titer data.

